warehouse/integrations/bigquery/bigquery_test.go

```diff
@@ -53,11 +54,15 @@ func TestIntegrationBigQuery(t *testing.T) {
 
 	t.Cleanup(func() {
 		for _, dataset := range []string{schema, sourcesSchema} {
-			require.NoError(t,
-				testhelper.WithConstantBackoff(func() (err error) {
-					return db.Dataset(dataset).DeleteWithContents(context.TODO())
-				}),
-				fmt.Sprintf("Failed dropping dataset %s for BigQuery", dataset),
+			require.Eventually(t, func() bool {
+				if err := db.Dataset(dataset).DeleteWithContents(context.TODO()); err != nil {
+					t.Logf("dropping dataset %s: %s", dataset, err.Error())
+					return false
+				}
+				return true
+			},
+				1*time.Minute,
+				1*time.Second,
```

Comment:
[optional] We could retry more often and potentially shorten the test time. If with a shorter interval the log above bothers you it can be removed or the test can be run withouth `-v`.

Same thing on the other `Eventually` below.

Reply:
I was biased with using `require.Eventually`. Let me use a simple for loop for retries.

Reply:
There is no need for Backoff and Eventually. In case of failures, we just need to retry a few of times.